tutorials/graph-theory/prims-algorithm.md

```diff
@@ -1,12 +1,378 @@
+        adjacency_list[current].append([abs(destination[0] - current[0]) + abs(destination[1] - current[1]), destination])
+        adjacency_list[destination].append([abs(destination[0] - current[0]) + abs(destination[1] - current[1]), current])
+```
+</TabItem>
+</Tabs>
+
+The adjacency list, will look something like this:
+```
+(0, 0): [[4, (2, 2)], [13, (3, 10)], [7, (5, 2)], [7, (7, 0)]]
+(2, 2): [[4, (0, 0)], [9, (3, 10)], [3, (5, 2)], [7, (7, 0)]]
+(3, 10): [[13, (0, 0)], [9, (2, 2)], [10, (5, 2)], [14, (7, 0)]]
+(5, 2): [[7, (0, 0)], [3, (2, 2)], [10, (3, 10)], [4, (7, 0)]]
+(7, 0): [[7, (0, 0)], [7, (2, 2)], [14, (3, 10)], [4, (5, 2)]]
+```
+
+The rest of the code follows an almost identical pattern as the previous one. The below being probably the only changes regarding to how the entire traversal is started.
+
+<Tabs>
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+
+```py
+# Here, our helper function contains one additional argument - the initial point (as here we don't need to randomly start)
+def min_cost_points(adjacency_list, initial):
+    # Number of nodes in the given tree
+    n = len(adjacency_list)
+    # A list we will use to implement a priority queue
+    connection_queue = []
+    # Total minimum cost to connect all points
+    res = 0
+    start = tuple(initial)
+    # To ensure we don't create closed loops
+    seen = set([start])
+```
+
+</TabItem>
+</Tabs>
+
+The below is the implementation of the main function. It is identical to the previous example in every way except for the final output (as in this question, we are guaranteed a solution)
+
+<Tabs>
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+  
+```py
+    for connection in adjacency_list[start]:
+        heapq.heappush(connection_queue, connection)
+
+    while len(seen) < n and connection_queue:
+        cost, current = heapq.heappop(connection_queue)
+        if current in seen:
+            continue
+
+        res += cost
+        seen.add(current)
+
+        for connection in adjacency_list[current]:
+            if connection[1] not in seen:
+                heapq.heappush(connection_queue, connection)
+
+    return res
+```
+</TabItem>
+</Tabs>
+
+#### Final Code:
+
+<Tabs>
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+
+```py
+import heapq
+
+class Solution:
+    def min_cost_points(self, adjacency_list, initial):
+        # Number of nodes in the given tree
+        n = len(adjacency_list)
+        # A list we will use to implement a priority queue
+        connection_queue = []
+        # Total minimum cost to connect all points
+        res = 0
+        # Taking the assumption that the n nodes in the adjacency list are from 1 to n
+        start = tuple(initial)
+        # To ensure we don't create closed loops
+        seen = set([start])
+
+        for connection in adjacency_list[start]:
```

Comment:
include comments for the below code like L314 - L323. (apply to other code block)